### PR TITLE
Make 'install' the default make target so that users must run 'make s…

### LINKED ISSUE #41: 'make' builds and runs which isn't always what you want. Ideally it should only run if you run 'make run' or 'make start'

### PR DESCRIPTION
**Pull request by snellers**
…tart' to start xania and doorman. Update README to reflect this along with some steps about gdb. Resolves #41

### CODE REVIEW

**Comment by snellers** [comment]
fyi @philjessies

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 .PHONY: default
-default: start
```
Comment (snellers) [inline Makefile:2]: You ok with this?

Comment (mattgodbolt) [reply Makefile:2]: 100%. There's a `make` way of setting the default target too (but I forgot about it when I wrote the makefile..) Purely FYI (and it's make 3.81 and above only):
```
.DEFAULT_GOAL := install
```

etc :)

File: README.md [hunk 2]
```diff
@@ -22,6 +22,13 @@ To build and run, type `make`:
 ```bash
 $ make
 ... lots of configuration nonsense here, downloads, cmake, and then compilation ...
+```
+
+Then to start it up:
+
+```bash
+$ make start
+
```
Comment (mattgodbolt) [inline README.md:31]: Maybe we could just change to `make start` on line 23 as that will build and start?

File: README.md [hunk 3]
```diff
@@ -74,8 +81,31 @@ will pick up on this and just use it. Command-line users can run `make reformat-
...
 There are some very early tests. To run the tests, use `make test`, or run them from your IDE.
 
+### Debugging Xania in GDB
+
+There's more than one way to do this, and one way is:  after building Xania you can run 
+it like so using a sub-shell in the `area` directory (use of `gdb` is optional):
+
+```bash
+$ (cd area && gdb ../install/bin/xania)
```
Comment (mattgodbolt) [inline README.md:94]: side note, I filed #53 to make this easier)

**Pull request merged**